### ISSUE TITLE
YOLOv8-Pose Custom Training

### ISSUE THREAD

**Issue by rabum** [ultralytics/ultralytics#16429]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Can I finetune the YOLOv8-Pose model with my own dataset with different keypoints setting? I see the model is pretrained on COCO Keypoints dataset with 17 keypoints, but I want to train a 7 keypoints model, how do I modify the setting and will the model perform badly because of the data conflict?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rabum, thank you for reaching out about YOLOv8-Pose custom training! 🚀 This is an automated response, but an Ultralytics engineer will assist you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com) for valuable insights into custom training, where you can find useful tips and methods for training your models 📚.

Since you're looking to finetune the YOLOv8-Pose model with a different number of keypoints, it's important to ensure your dataset is appropriately formatted. If you encounter a bug during this process, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us assist you more effectively.

For custom training ❓ questions, detailed information such as dataset examples and training logs will be very helpful. Make sure you are also following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our vibrant Ultralytics community for more support. Engage in real-time discussions on [Discord](https://ultralytics.com/discord) 🎧, enjoy in-depth conversations on [Discourse](https://community.ultralytics.com), or connect on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you are using the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can be run in these environments with all necessary dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

The current CI badge status above indicates if all tests are passing. Tests include verifying correct operation of all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across various platforms. 

Happy training! 🌟

**Comment by Y-T-G** [comment]
You can train on your dataset. But it won't be fine-tuning. It will be transfer learning.

You can follow the datasets guide to set up your yaml.

https://docs.ultralytics.com/datasets/pose/

**Comment by rabum** [comment]
> You can train on your dataset. But it won't be fine-tuning. It will be transfer learning.
> 
> You can follow the datasets guide to set up your yaml.
> 
> https://docs.ultralytics.com/datasets/pose/

I see, so I need to set up kpt_shape and flip_idx, but is there a mapping to it? For example, my kpt_shape may be:
[5, 3]
the five keypoints are [left eye, right eye, mouth, left hand, right hand], do I need to set flip_idx to [1, 0, 2, 4, 3]? I don't really understand the meaning

**Comment by Y-T-G** [comment]
Yes, your `kpt_shape` and `flip_idx` is correct. There's no mapping, as in you can't add names for them. But in your labels, they should be in the correct order.
